web/src/peer_data.ts

```diff
@@ -453,3 +474,63 @@ export async function get_unique_subscriber_count_for_streams(
     }
     return valid_subscribers.size;
 }
+
+// This function might retry up to 5 times to fetch data, then will
+// give up. `num_attempts` should only be set in the recursive call.
+async function load_subscriptions_for_user_with_retry(
+    user_id: number,
+    num_attempts = 0,
+): Promise<void> {
+    const subscriptions_promise = (async () => {
+        let subscriptions: number[];
+        try {
+            const result = await channel.get({
+                url: `/json/users/${user_id}/channels`,
+            });
+            subscriptions =
+                fetch_user_subscriptions_response_schema.parse(result).subscribed_channel_ids;
```

Comment:
I don't love that this error-handling logic is treating a 400 error (like auth denied) as a thing that should be retried. We have standard error-handling patterns that seem appropriate here.

Reply:
What do you think about adding a `error_retry` callback to `channel.get` call which only gets called if the error should be retried? So, instead of `error` we would define `error_retry` here.